circle.yml

```diff
@@ -29,14 +29,20 @@ test:
 
 deployment:
   production:
-    branch: [master]
+    branch: master
     commands:
       - cf login -a https://api.cloud.gov -u gsa-wds_deployer -p $CF_GSA_WDS_PASS -o gsa-wds -s wds-production
       - cf zero-downtime-push wds-microsite -f manifest.yml -p _site
       - cf map-route wds-microsite standards.usa.gov
   staging:
-    branch: [staging]
+    branch: staging
     commands:
       - cf login -a https://api.cloud.gov -u gsa-wds_deployer -p $CF_GSA_WDS_PASS -o gsa-wds -s wds-staging
       - cf zero-downtime-push wds-microsite -f manifest-staging.yml -p _site
       - cf map-route wds-microsite standards-staging.usa.gov
+  demo:
+    branch: /demo_.*/
+    commands:
+      - cf login -a https://api.cloud.gov -u gsa-wds_deployer -p $CF_GSA_WDS_PASS -o gsa-wds -s wds-demo
+      - cf zero-downtime-push wds-microsite -f manifest-demo.yml -p _site
+      - cf map-route wds-microsite standards-demo.usa.gov
```

Comment:
Might want to consider modifying L47 & L48 to be tied to the branch being pushed up.

```shell
# no need for zero-downtime as it consumes 2x resources.
- cf push "wds-microsite-$(CIRCLE_BRANCH)" -f manifest-demo.yml -p _site
- cf map-route "wds-microsite-$(CIRCLE_BRANCH)" "standards-demo-$(CIRCLE_BRANCH).usa.gov"
```

This would allow for multiple PRs for demoing, so I would include some documentation on spinning those instances down. Maybe something like...

---

### Continuously Delivered Demo Sites

Submitting a PR with a branch name prefixed with `demo_` will trigger a deployment to `standards-demo-<BRANCH_NAME>.usa.gov`. Please be mindful of using these deployments sparingly. Once a branch has been deleted or abandoned, the instance in :cloud:.gov can be deleted with the following commands.

```shell
$ cf t -o gsa-wds -s wds-demo # target the proper org and space.
$ cf apps # lists all applications in the space.
$ cf delete -f standards-demo-<BRANCH_NAME> # to delete the application without prompting.
```

This feature is limited to pull requests not coming from forks.